src/realm/integer_flex_compressor.hpp

```diff
@@ -189,116 +190,99 @@ inline bool FlexCompressor::find_all(const Array& arr, int64_t value, size_t sta
-    else if constexpr (std::is_same_v<Less, Cond>) {
-        return do_find_all<Less, GreaterEqual, Less>(arr, value, start, end, baseindex, state);
-    }
-    else if constexpr (std::is_same_v<Greater, Cond>) {
-        return do_find_all<Greater, Greater, GreaterEqual>(arr, value, start, end, baseindex, state);
-    }
-    return true;
-}
-
-template <typename LinearCond, typename VectorCond1, typename VectorCond2>
-inline bool FlexCompressor::do_find_all(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
-                                        QueryStateBase* state)
-{
-    const auto v_width = arr.m_width;
-    const auto v_range = arr.integer_compressor().v_size();
-    const auto ndx_range = end - start;
-    if (!run_parallel_subscan<LinearCond>(v_width, v_range, ndx_range))
-        return find_linear<LinearCond>(arr, value, start, end, baseindex, state);
-    return find_parallel<VectorCond1, VectorCond2>(arr, value, start, end, baseindex, state);
-}
-
-template <typename Cond>
-inline bool FlexCompressor::find_linear(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
-                                        QueryStateBase* state)
-{
-    const auto cmp = [](int64_t item, int64_t key) {
-        if constexpr (std::is_same_v<Cond, Equal>)
-            return item == key;
-        if constexpr (std::is_same_v<Cond, NotEqual>)
-            return item != key;
-        if constexpr (std::is_same_v<Cond, Less>)
-            return item < key;
-        if constexpr (std::is_same_v<Cond, Greater>)
-            return item > key;
-        REALM_UNREACHABLE();
-    };
-
-    const auto& c = arr.integer_compressor();
-    const auto offset = c.v_width() * c.v_size();
-    const auto ndx_w = c.ndx_width();
-    const auto v_w = c.v_width();
-    const auto data = c.data();
-    const auto mask = c.v_mask();
-    BfIterator ndx_iterator{data, offset, ndx_w, ndx_w, start};
-    BfIterator data_iterator{data, 0, v_w, v_w, static_cast<size_t>(*ndx_iterator)};
-    while (start < end) {
-        const auto sv = sign_extend_field_by_mask(mask, *data_iterator);
-        if (cmp(sv, value) && !state->match(start + baseindex))
-            return false;
-        ndx_iterator.move(++start);
-        data_iterator.move(static_cast<size_t>(*ndx_iterator));
-    }
-    return true;
-}
-
-template <typename VectorCond1, typename VectorCond2>
-inline bool FlexCompressor::find_parallel(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
-                                          QueryStateBase* state)
-{
-    //
-    // algorithm idea: first try to find in the array of values (should be shorter in size but more bits) using
-    // VectorCond1.
-    //                 Then match the index found in the array of indices using VectorCond2
-    //
-
     const auto& compressor = arr.integer_compressor();
-    const auto v_width = compressor.v_width();
+    const auto v_width = arr.m_width;
```

Comment:
NIT: find_all now contains both find value and finds index, and we do that for all type of comparators, nice! I would have split find_all in `find_value` and `find_index`, just to make the code more readable. But It is up to you.